ui/page/receive_page.go

```diff
@@ -137,6 +138,16 @@ func (pg *ReceivePage) OnNavigatedTo() {
 	pg.selector.ListenForTxNotifications(pg.ctx)
 	pg.selector.SelectFirstWalletValidAccount(nil) // Want to reset the user's selection everytime this page appears?
 	// might be better to track the last selection in a variable and reselect it.
+	selectedWallet := pg.multiWallet.WalletWithID(pg.selector.SelectedAccount().WalletID)
+	currentAddress, err := selectedWallet.CurrentAddress(pg.selector.SelectedAccount().Number)
+	if err != nil {
+		log.Errorf("Error getting current address: %v", err)
+		pg.Toast.NotifyError(fmt.Sprintf("Error getting current address: %v", err))
+	} else {
+		pg.currentAddress = currentAddress
+	}
+
+	pg.generateQRForAddress()
```

Comment:
although not related to your PR it makes sense to modify this method to accept an address as an argument, in which it would be used to generate the QR code, doesn't make much sense to update the address separately and generate the QR code separately when they are directly related

or to avoid feature creeping, an issue can be created and it can be done in another PR

Reply:
I think this approach works for me though. Its really not too much of an issue and the address are used from different methods.